lib/sidekiq/instrument/mixin.rb

```diff
@@ -8,8 +8,14 @@ def metric_name(worker, event)
       end
     end
 
-    def worker_dog_options(worker)
-      { tags: ["queue:#{queue_name(worker)}", "worker:#{underscore(class_name(worker))}"] }
+    def worker_dog_options(worker, job)
+      tags = job.dig('tags') || []
+      {
+        tags: [
+          "queue:#{queue_name(worker)}",
+          "worker:#{underscore(class_name(worker))}"
+        ].concat(tags)
+      }
```

Comment:
nit
```suggestion
      {
        tags: [
          "queue:#{queue_name(worker)}",
          "worker:#{underscore(class_name(worker))}"
        ].concat(job.fetch('tags', []))
      }
```

Reply:
btw, is there a chance job tags overwrite the existing tags? If so, which one has precedence over the other?

Reply:
What do you mean by "existing tags"? The queue and worker ones inside this method, or some other source?